.github/workflows/check-migrations.yml

```diff
@@ -60,21 +60,43 @@ jobs:
 
       - name: Build ${{ matrix.runtime.name }}
         run: |
-          cargo build --profile production -p ${{ matrix.runtime.package }} --features try-runtime
+          cargo build --profile production -p ${{ matrix.runtime.package }} --features try-runtime -q --locked
 
       - name: Check migrations
         run: |
           PACKAGE_NAME=${{ matrix.runtime.package }}
           RUNTIME_BLOB_NAME=$(echo $PACKAGE_NAME | sed 's/-/_/g').compact.compressed.wasm
+          RUNTIME_BLOB_PATH=./target/production/wbuild/$PACKAGE_NAME/$RUNTIME_BLOB_NAME
+
           # When running on relay, we don't need weight checks.
           NO_WEIGHT_WARNINGS_FLAG=""
           if [[ "${{ matrix.runtime.is_relay }}" == "true" ]]; then
             NO_WEIGHT_WARNINGS_FLAG="--no-weight-warnings"
           fi
-          # Disable idempotency checks for now because the Scheduler pallet MigrateToV1 migration is
-          # non-idempotent at its root in Polkadot SDK V1.2.0. In V1.3.0 we can re-enable 
-          # idempotency checks.
-          EXTRA_ARGS="--disable-spec-version-check --disable-idempotency-checks"
+
+          # Disable the spec version check when we dont want to release.
+          EXTRA_FLAGS=""
+          if ! .github/changelog-processor.py CHANGELOG.md --should-release ; then
+            EXTRA_FLAGS="--disable-spec-version-check"
+          fi
+
+          # First create a snapshot that we can re-use for the two other commands:
+          ./try-runtime \
+            create-snapshot --uri ${{ matrix.runtime.uri }} \
+            onchain.snap
+          
+          # Check with pre-and-post
+          ./try-runtime \
+            --runtime $RUNTIME_BLOB_PATH \
+            on-runtime-upgrade --checks=pre-and-post \
+            $EXTRA_FLAGS $NO_WEIGHT_WARNINGS_FLAG \
+            snap -p onchain.snap
+        
+          # Optional: also check try-state invariants
```

Comment:
How is this optional? And why not just directly do it on the first run?

Reply:
I am trying it out for now, could be that it fails and in that case i still wanted to see that the `pre-and-post` works.

Reply:
I think `try-runtime-cli` is already doing these checks separately.

Reply:
yea maybe. I will remove it that we can merge. Seems like maybe we should merge your MR first since there a non-idempotent migration in polkadot.